azurerm/resource_arm_public_ip.go

```diff
@@ -122,6 +135,13 @@ func resourceArmPublicIpCreate(d *schema.ResourceData, meta interface{}) error {
 
 	idleTimeout := d.Get("idle_timeout_in_minutes").(int)
 	ipAllocationMethod := network.IPAllocationMethod(d.Get("public_ip_address_allocation").(string))
+	ipVersion := network.IPVersion(d.Get("ip_version").(string))
+
+	if strings.ToLower(string(ipVersion)) == "ipv6" {
+		if strings.ToLower(string(ipAllocationMethod)) == "static" {
```

Comment:
_minor_ rather than lower-casing it we can use `strings.EqualsFold(string(ipAllocationMethod), "static")` here to use a case-invarient check

Reply:
Am making the change. But I am curious, is there a particular reason for this preference?

Reply:
heh sorry I missed out using the Enum here - in general we'd tend to do something like:

```
strings.EqualsFold(string(ipAllocationMethod), network.Static)
```

although in this specific case (if we remove the case sensitivity) - we should actually be able to do a strict equality comparison I guess? 🤔

Reply:
@Krueladin sorry, that's a typo - I've updated the example